src/libs/actions/Policy/Tag.ts

```diff
@@ -583,10 +583,61 @@ function setPolicyRequiresTag(policyID: string, requiresTag: boolean) {
     API.write(WRITE_COMMANDS.SET_POLICY_REQUIRES_TAG, parameters, onyxData);
 }
 
+function setPolicyTagsRequired(policyID: string, requiresTag: boolean, tagListIndex: number) {
+    const policyTag = PolicyUtils.getTagLists(allPolicyTags?.[`${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`] ?? {})?.[tagListIndex] ?? {};
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
+                value: {
+                    [policyTag.name]: {
+                        required: requiresTag,
+                        pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
```

Comment:
```suggestion
                        pendingFields: {required: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE},
```
We are using pendingFields to for the offline indicator and this action is `update`

Reply:
https://github.com/Expensify/App/pull/42972/commits/8b777f11221410ff117cb8e43137f09213ab4024